src/main/java/graphql/schema/MapDataFetcher.java

```diff
@@ -0,0 +1,22 @@
+package graphql.schema;
+
+import java.util.Map;
+
+import static graphql.Assert.assertNotNull;
+
+public class MapDataFetcher implements DataFetcher {
+
+    private final String keyName;
+
+    public MapDataFetcher(String keyName) {
+        assertNotNull(keyName, "`keyName` can't be null");
+        this.keyName = keyName;
+    }
+
+    @Override
+    public Object get(DataFetchingEnvironment env) {
+        Object source = env.getSource();
+        if (source == null || !(source instanceof Map)) return null;
+        return ((Map<?, ?>) source).get(keyName);
```

Comment:
@bpatters yeah I was confused about that too (that's how it was in the current `PropertyDataFetcher` and `FieldDataFetcher`s).

> So you might as well cast the Map to a Map

So, like this?

```
return ((Map) source).get(keyName);
```

Per your comment and my confusion, why not:

```
return ((Map<String, ?>) source).get(keyName);
```